.test-infra/kubernetes/hadoop/SmallITCluster/hdfs-single-datanode-cluster.yml

```diff
@@ -0,0 +1,76 @@
+#    Licensed to the Apache Software Foundation (ASF) under one or more
+#    contributor license agreements.  See the NOTICE file distributed with
+#    this work for additional information regarding copyright ownership.
+#    The ASF licenses this file to You under the Apache License, Version 2.0
+#    (the "License"); you may not use this file except in compliance with
+#    the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: hadoop
+  labels:
+    name: hadoop
+spec:
+  ports:
+    - name: sshd
+      port: 2122
+    - name: hdfs
+      port: 9000
+    - name: web
+      port: 50070
+    - name: datanode
+      port: 50010
+    - name: datanode-icp
+      port: 50020
+    - name: datanode-http
+      port: 50075
```

Comment:
Any reason why these specific port values were chosen ?

Reply:
These ports are standard hdfs datanode ports, I didn't want to change them. They are briefly described here: http://blog.cloudera.com/blog/2009/08/hadoop-default-ports-quick-reference/. If you are asking in terms of security the most secure working setup would be with only two ports exposed 9000 and 50010, however then we are disabling external access via http to namenode console on port 50070 which gives nice debug output. This could be really useful for users playing with hdfs for the first time.

Reply:
Sounds good.